### PR TITLE
Attach - Improve IR Strobe blinking pattern and increase lifetime

### LINKED ISSUE #4946: IR Strobe detaches after time interval
**Arma 3 Version:** `1.66` (stable)
**CBA Version:** `3.1.2` (stable)
**ACE3 Version:** `3.9.0` (stable)

**Mods:**
- `@CBA_A3`
- `@ace`

**Description:**
The time seems to be around 10 minutes.
Bux said it might be the lifeTime of the strobe. But it still continues blinking after it detaches.
I sure sounds to me like the batteries of the strobe ran out and it should stop blinking. But it continues blinking for atleast 5 minutes.

After it detaches 
`player getVariable ["ace_attach_attached",true]`
returns `[[<NULL-object>,"B_IR_Grenade"]]`

**Steps to reproduce:**
- Attach vanilla Nato IR strobe
- Go afk for 10-20 minutes
- Move a few steps away and the IR strobe will be floating in the air behind you

**Where did the issue occur?**
- Dedicated

### PR DESCRIPTION
**Pull request by mharis001**
**When merged this pull request will:**
- Change IR Strobe blinking pattern to 0.1 s on, 0.9 s off
- Increase lifetime to 8 hours
- Close #4946 
- Replace #5233

### CODE REVIEW

File: addons/attach/CfgVehicles.hpp [hunk 1]
```diff
@@ -86,6 +86,10 @@ class CfgVehicles {
             dayLight = 0;
             onlyInNvg = 1;
             useFlare = 0;
+            maxLifetime = 28800; // 8 hours
```
Comment (commy2) [inline addons/attach/CfgVehicles.hpp:89]:
```suggestion
            maxLifetime = 8 * 60 * 60;
```
?
Does the engine even respect high lifetimes on this simulation type?

Comment (mharis001) [reply addons/attach/CfgVehicles.hpp:89]: It should in the next update. https://feedback.bistudio.com/T158909

**Comment by PabstMirror** [comment]
Adding different strobes with custom patterns would be neat for IFF
also synchronized https://www.youtube.com/watch?v=bR_z-6m7g1Q
but I'm not sure how

File: addons/attach/CfgVehicles.hpp (hunk 1 above)
Comment (BaerMitUmlaut) [reply addons/attach/CfgVehicles.hpp]: Shouldn't this be quoted?

**Comment by mharis001** [comment]
We could provide a couple variants with different patterns. Simplest solution would be to add additional strobe items for each of the patterns but could also allow the user to select the pattern using actions. Not sure how to handle synchronized strobes or how feasible that is.

**Comment by BrettMayson** [comment]
If we delete all the strobes that are wanted to be synchronized and recreate them all at once, would that do the trick?

**Comment by BaerMitUmlaut** [comment]
Just attach local copies.

**Comment by TheCandianVendingMachine** [comment]
What about avoiding creating the strobe until `sync_time_start % CBA_missionTime == 0`?

File: addons/attach/CfgVehicles.hpp [hunk 2]
```diff
@@ -86,6 +86,10 @@ class CfgVehicles {
             dayLight = 0;
             onlyInNvg = 1;
             useFlare = 0;
+            maxLifetime = 8 * 60 * 60;
```
Comment (BaerMitUmlaut) [inline addons/attach/CfgVehicles.hpp:89]:
```suggestion
            maxLifetime = "8 * 60 * 60";
```

File: addons/attach/CfgVehicles.hpp [hunk 3]
```diff
@@ -86,6 +86,10 @@ class CfgVehicles {
...
+            blinkingPattern[] = {0.1, 0.9}; // 0.1 s on, 0.9 s off
```
Comment (BaerMitUmlaut) [inline addons/attach/CfgVehicles.hpp:90]:
```suggestion
            blinkingPattern[] = {0.1, 1.1}; // 0.1 s on, 1.1 s off
```

According to the [MS-2000 manual](http://www.ostrich.co.jp/wps/wp-content/uploads/2016/11/manual-y1-03-0275b.pdf) the blink rate should be 50±10 flashes per minute. I've checked out your branch and I have to say I prefer `blinkingPattern[] = {0.1, 1.1};`, which would meet that spec, seems a little less "nervous".

File: addons/attach/CfgVehicles.hpp [hunk 4]
```diff
@@ -86,6 +86,10 @@ class CfgVehicles {
...
+            blinkingStartsOn = 1;
+            blinkingPatternGuarantee = 1;
```
Comment (BaerMitUmlaut) [inline addons/attach/CfgVehicles.hpp:92]: This property doesn't seem to work currently, you get no blinking at all if you use an on time less than a frame's duration (@dedmen?). But it doesn't hurt adding it.

**Review by BaerMitUmlaut — CHANGES_REQUESTED** [review]

**Comment by BaerMitUmlaut** [comment]
We can do additional features like synched blinking later, I would prefer to get this merged for the next release. Unless @mharis001  is already working on that?

**Comment by mharis001** [comment]
I briefly experimented with adding strobes with different patterns but that can be done later. Let's get this merged for the next release.

**Review by BaerMitUmlaut — APPROVED** [review]

**Comment by PabstMirror** [comment]
Safe to merge now, but blinking won't take effect til 2.06?

**Review by PabstMirror — APPROVED** [review]

**Comment by mharis001** [comment]
Yes safe to merge now. These config entries will not be used until 2.06.

File: addons/attach/CfgVehicles.hpp (hunk 4 above)
Comment (dedmen) [reply addons/attach/CfgVehicles.hpp:92]: Dunno, should be same as for that other blinking thing :D

**Pull request merged**